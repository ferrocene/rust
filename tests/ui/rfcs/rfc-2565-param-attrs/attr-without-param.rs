//@ revisions: edition2015 postedition2015
//@[edition2015] edition: 2015
//@[postedition2015] edition: 2018..

#[cfg(false)]
impl S {
    fn f(#[attr]) {} //~ ERROR expected parameter name, found `)`
}

#[cfg(false)]
impl T for S {
    fn f(#[attr]) {}
    //~^ ERROR expected parameter name, found `)`
}

#[cfg(false)]
trait T {
    fn f(#[attr]);
    //[edition2015]~^ ERROR expected argument name, found `)`
    //[postedition2015]~^^ ERROR expected parameter name, found `)`
}

fn main() {}
