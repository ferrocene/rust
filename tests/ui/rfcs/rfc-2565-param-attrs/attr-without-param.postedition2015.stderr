error: expected parameter name, found `)`
  --> $DIR/attr-without-param.rs:7:17
   |
LL |     fn f(#[attr]) {}
   |                 ^ expected parameter name

error: expected parameter name, found `)`
  --> $DIR/attr-without-param.rs:12:17
   |
LL |     fn f(#[attr]) {}
   |                 ^ expected parameter name

error: expected parameter name, found `)`
  --> $DIR/attr-without-param.rs:18:17
   |
LL |     fn f(#[attr]);
   |                 ^ expected parameter name

error: aborting due to 3 previous errors

